### PR TITLE
email embed pages inform parent when page loads

### PR DESCRIPTION
**Pull request by dblatcher**
## What does this change?
Adds function to the page template for email - embed form pages to post a message to the parent window to inform it when a new page loads - I.E. when the form page changes to the "success" or "failure" page.

This is to help address a colour contrast issue in sign-up thrashers. Although the "thrasher" versions form pages can be styled to fit the background used in the thrasher (parent window) - I.E. the form can be styled have white text to contrast with a dark background-  there is only one version of the "success" or "failure" page and always uses black text. 

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots (issue we intend to address)

issue - low contrast on the dark coloured container:
Screenshot 2022-10-17 at 18 13 47

## What is the value of this and can you measure success?
Parent window loading using the forms in iframes will be able to detect when the iframe location change to the success or failure page.

The first application will be for thrashers using dark backgrounds to apply styling to the success / failure pages to improve colour contrasts.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist
Although, this PR is intended to resolve a colour contrast issue, it does not make visual changes itself - the thrashers will make those.

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: common/app/views/emailEmbed.scala.html [hunk 1]
```diff
@@ -49,7 +49,15 @@
...
 
-            revealPrivacyWordingWhenUserInteractsWithEmailField()
+            function informParentOfLoad(loadEvent) {
+                window.parent.postMessage({
+                    subject: 'emailEmbedPageLoaded',
+                    path: window.location.pathname,
+                }, window.origin);
+            }
+
+            window.addEventListener('DOMContentLoaded', informParentOfLoad);
```
Comment (mxdvl) [inline common/app/views/emailEmbed.scala.html:59]: Remeber, functions are hoisted in JS, so you can register your listener higher up.

File: common/app/views/emailEmbed.scala.html [hunk 2]
```diff
@@ -49,7 +49,15 @@
...
+            revealPrivacyWordingWhenUserInteractsWithEmailField();
```
Comment (mxdvl) [inline common/app/views/emailEmbed.scala.html:60]: Could we keep the existing line as-is, removing the trailing `;`?

**Review by mxdvl — APPROVED** [review]
LGTM

File: common/app/views/emailEmbed.scala.html (hunk 2 above)
Comment (dblatcher) [reply common/app/views/emailEmbed.scala.html:60]: ok, will do.

**Pull request merged**